src/js/utils/html-utils.js

```diff
@@ -75,20 +95,40 @@
          *   );
          *
          * @param {string} formatString The string to be interpolated.
-         * @param {Object} parameters An optional set of parameters to the template.
-         * @returns {string} The resulting safely escaped string flagged as
-         * an HTML snippet.
+         * @param {Object} parameters An optional set of parameters for interpolation.
+         * @returns {HtmlSnippet} The resulting safely escaped HTML snippet.
          */
         interpolateHtml = function(formatString, parameters) {
             var result = StringUtils.interpolate(
-                escape(formatString).toString(),
-                _.mapObject(parameters, escape)
+                ensureHtml(formatString).toString(),
+                _.mapObject(parameters, ensureHtml)
             );
             return HTML(result);
         };
 
         /**
-         * Returns a function that renders an Underscore template as a string.
+         * Joins multiple strings and/or HTML snippets together to produce
+         * a single safely escaped HTML snippet.
+         *
+         * For each item, if it is provided as an HTML snippet then it is joined
+         * directly. If the item is a string then it is assumed to be unescaped and
+         * so it is first escaped before being joined.
+         *
+         * @param {...(string|HtmlSnippet)} items The strings and/or HTML snippets
+         * to be joined together.
+         * @returns {HtmlSnippet} The resulting safely escaped HTML snippet.
+         */
+        joinHtml = function() {
+            var html = '',
+                i;
+            for (i = 0; i < arguments.length; i++) {
```

Comment:
Nitpick: the length should be assigned to a variable instead of calculating length on each iteration.